src/main/java/tommy/exception/TommyException.java

```diff
@@ -0,0 +1,24 @@
+package tommy.exception;
+
+/**
+ * Abstract class that is the parent class of all exceptions created in the program.
+ */
+public abstract class TommyException extends Exception {
+
+    /**
+     * Constructor for an TommyException,
+     * which initialises the exception with its error message.
+     *
+     * @param message Description of the TommyException error.
+     */
+    public TommyException(String message) {
+        super(message);
+    }
+
+    /**
+     * Prints the descriptive message of the exception error for the user.
+     *
+     * @return Descriptive error message.
+     */
+    public abstract String errorMessage();
```

Comment:
Perhaps you could change the method name to a verb here.

```suggestion
    public abstract String getErrorMessage();
```